summingbird-online/src/main/scala/com/twitter/summingbird/online/executor/AsyncBase.scala

```diff
@@ -107,13 +136,15 @@ abstract class AsyncBase[I, O, S, D, RC](maxWaitingFutures: MaxWaitingFutures, m
     if (pendingFuturesCount > maxWaitingFuturesCount) {
       // Too many futures waiting, let's clear.
       outstandingFutures.dequeueAll(_.isDefined)
```

Comment:
I wrote a benchmark for a case with high store latency. I found that this change clearly increases throughput. Also the watermarks didn't make any difference. I've removed the watermark logic.

Here are the benchmark numbers before and after this change:
Before
SummerBenchmark.highLatencyStore  thrpt   10  **2077.829 ± 289.029  ops/s**

After
SummerBenchmark.highLatencyStore  thrpt   10  **3267.906 ± 139.766  ops/s**

I ran the benchmark a number of times and found the results to be consistent. 

Set up was:
Summer - SyncSummingQueue - buffersize 100, flush frequency 1 sec
MaxWaitingFutures - 1000
MaxFutureWaitTime - 5 sec
MaxEmitsPerExecute - Int.Max

The benchmark is a big change, I'm planning to create a separate PR for it after a couple weeks.
@johnynek PTAL